lib/anycable/nsq/handler.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+require 'anycable/socket'
+
+module AnyCable
+  module NSQ
+    class Handler
+      def handle(data)
+        request = JSON.parse(data)
+        case request['command']
+        when "connect"
+          connect(request)
+        when "disconnect"
+        when "perform", "subscribe", "unsubscribe"
+          perform(request)
+        end
+      end
+      # Handle connection request from WebSocket server
+      def connect(request)
+        logger.debug("RPC Connect: #{request}")
+
+        socket = build_socket(env: rack_env(request))
+        connection = factory.call(socket)
+
+        connection.handle_open
+        if socket.closed?
+          AnyCable.to_client.write({
+            "ws_id" => request["ws_id"],
+            "command" => request["command"],
+            "status" => "failure"
+          }.to_json)
+        else
+
+          AnyCable.to_client.write({
+            "ws_id" => request["ws_id"],
+            "command" => request["command"],
+            "status" => "success",
+            "identifiers" => connection.identifiers_json,
+            "transmissions" => socket.transmissions
+          }.to_json)
+        end
+      end
+
+      def disconnect(request)
+        logger.debug("RPC Disonnect: #{request}")
+
+        socket = build_socket(env: rack_env(request))
+
+        connection = factory.call(
+          socket,
+          identifiers: request.identifiers,
+          subscriptions: request.subscriptions
+        )
+
+        if connection.handle_close
+          AnyCable::DisconnectResponse.new(status: AnyCable::Status::SUCCESS)
+        else
+          AnyCable::DisconnectResponse.new(status: AnyCable::Status::FAILURE)
+        end
+      end
+
+      def perform(message)
+        logger.debug("RPC Command: #{message}")
+        msg = JSON.load(message["message"])
+        socket = build_socket(env: rack_env(message))
+
+        connection = factory.call(
+          socket,
+          identifiers: msg["identifier"]
+        )
+        message["command"] = "message" if message["command"] == "perform"
+        begin
+          #binding.pry
```

Comment:
probably want to just delete this i would think

Reply:
Same with this... I'll clean it up!